swap_meet/item.py

```diff
@@ -18,12 +18,23 @@ def __str__(self):
         return f"An object of type {self.get_category()} with id {self.id}."
 
     def condition_description(self):
-        descriptions = {
-        0: "You probably want a glove for this one...",
-        1: "Heavily used",
-        2: "Worn but usable",
-        3: "Gently used",
-        4: "Like new",
-        5: "Mint condition"
-    }
-        return descriptions.get(self.condition)
+    #     descriptions = {
+    #     0: "You probably want a glove for this one...",
+    #     1: "Heavily used",
+    #     2: "Worn but usable",
+    #     3: "Gently used",
+    #     4: "Like new",
+    #     5: "Mint condition"}
+    #     return descriptions.get(self.condition)
+        if self.condition <= 1:
+            return "You probably want a glove for this one..."
+        elif 1 < self.condition <= 2:
+            return "Heavily used"
+        elif 2 < self.condition <= 3:
+            return "Worn but usable"
+        elif 3 < self.condition <= 4:
+            return "Gently used"
+        elif 4 < self.condition <= 5:
+            return "Like new"
+        else:
+            return "Mint condition"
```

Comment:
Notice that in a change set of ranges like this, we don't need to repeat the end of the range that was checked in a previous condition.
```py
        if self.condition <= 1:
            return "You probably want a glove for this one..."
        elif self.condition <= 2:  # doesn't need to check that the value is greater than 1 here
            return "Heavily used"
        ...  # other cases
        else:
            return "Mint condition"
```

Reply:
This revised approach does handle float cases properly, but from a code structure perspective, I prefer the previous dictionary-based approach. Separating "configuration" information from the logic is almost always preferred for maintainability and understandability (it's another aspect of separation of concerns/single responsibility principle), while mixing data and logic as we need to do with a chained `if`/`else` can become brittle over time.

Can you think of a way to adapt the dictionary approach to be able to work with floats? We can't use ranges as the keys of a dictionary, but is there a way we can "squash" say all the numbers from 0 up to juuuuust below 1 down to the 0 key? Maybe it could also squash everything from 1 up to just below 2 down the 1 key?